### PR TITLE
Add Helpline Select on ZM offline contact form

### PR DESCRIPTION
**Pull request by murilovmachado**
Jira: https://bugs.benetech.org/browse/CHI-633

This PR adds a **Helpline** select on **Contact Info** Tab when creating an **offline contact** on **ZM**.

P.S.: The easiest way to test this is to edit **HrmFormPlugin** line 68 to be `definitionVersion: 'v1',` because **dev** environment is configured on Twilio to use ZA definitions. 

image

### CODE REVIEW

File: plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json:1]: It feels a little weird to me that this tab is part fixed (via `ContactlessTaskTabDefinition`) and part based on a configurable `formDefinition`.  Is that going to cause trouble for us?  I am okay with it if both of you are, but I want to ask.

File: plugin-hrm-form/src/services/ContactService.ts [hunk 2]
```diff
@@ -193,13 +193,15 @@ export async function saveToHrm(task, form, workerSid, helpline, uniqueIdentifie
    */
   const formToSend = transformForm(rawForm);
 
+  const helplineToSend = (isOfflineContactTask(task) && form.contactlessTask.helpline) || helpline;
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/services/ContactService.ts:196]: Should we rename the `helpline` function parameter to be `workerHelpline` or something similar to make it clear where it's coming from?

Comment (nickhurlburt) [inline plugin-hrm-form/src/services/ContactService.ts:196]: Is there any risk that `form.contactlessTask` would be `undefined`, causing an error here?  If maybe we're using the `za-v1` `formDefinition`?  Would `form.contactlessTask?.helpline` be safer?

File: plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  {
+    "name": "helpline",
+    "label": "Helpline",
+    "type": "select",
+    "options": [
+      { "value": "Unknown", "label": "" },
+      { "value": "ChildLine", "label": "ChildLine" },
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json:8]: Do we have a way to set a default value?  If so, then check with Dee if we can set the default to "ChildLine", at least while we're not running LifeLine on Aselo.

**Review by nickhurlburt — COMMENTED** [review]
Good, with comments.  I am a little nervous about creating our first hybrid tab that has some fixed fields and some helpline-specific configuration, since it's a new pattern.  I'm fine with it if you both can't see any problems with it.

File: plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json (hunk 1 above)
Comment (murilovmachado) [reply plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json:1]: The main difference between the `ContactlessTaskTab` and the other tabs is that we have **Counselors** select, which is dynamic. Because of that, we can't use plain JSON files to define this tab, we need typescript code. This way, we could have the file `ContactlessTaskTab.ts` instead of `ContactlessTaskTab.json` (per helpline, ex: v1, za-v1) defining all fields for this tab.

The tradeoff with having typescript files:
- May be harder for other people not familiar with the code to edit it;
- May impact our idea of having these JSON definition files living outside of our codebase;

That's why I've ended up implementing a hybrid approach:
- We still have a typescript that handles counselors select (and other common fields)
- But other custom fields are defined at `ContactlessTaskTab.json`, so other people can edit it easily;

The drawbacks of the current implementation are:
- It's a new approach for this tab only, which hurts standardization;
- Currently, we can't change the order of fields for each helpline;

One idea that may solve our problems of having a dynamic field being defined in static JSONs would be to create a new form type called `counselorSelect`, which we would use like this:
```
{
      name: 'counselors',
      type: 'counselorSelect',
      label: 'Counsellor',
      required: { value: true, message: 'RequiredFieldError' },
},
```
Another idea that may bit a little bit harder to implement would be to have `initialOptions` the same way we have `initialValues`, like this:
```
const counselorOptions = getCounselorOptions();
const initialOptions = { counselors: counselorOptions };

<ContactlessTaskTab
      task={task}
      display={subroute === 'contactlessTask'}
      definition={currentDefinitionVersion.tabbedForms.ContactlessTaskTab}
      initialValues={contactForm.contactlessTask}
      initialOptions={initialOptions}
/>
```
With this in hand, we can have any number of dynamic select fields.

-----

In summary there're, at least, 5 different ways to handle this:
- Customizing with typescript files (instead of JSON);
- Hybrid: only the custom fields defined in JSON (this PR);
- Create `counselorSelect` specific type
- Implement `initialOptions`;
- Hardcode ;

What do you guys think would be better for now? @nickhurlburt @GPaoloni

Comment (nickhurlburt) [reply plugin-hrm-form/src/formDefinitions/v1/tabbedForms/ContactlessTaskTab.json:1]: This is an excellent description of the issue and the possible solutions, thank you!  Given this description, which we can refer back to later, using a hybrid model for this tab sounds good to me.  If new needs come up in the future, we can revisit this description and decide if the other options make more sense at that time.

**Review by nickhurlburt — APPROVED** [review]
Looks great, thanks!

**Comment by nickhurlburt** [comment]
@murilovmachado looks like this got merged to `staging`.  Would you mind merging staging into master and then just deleting `staging`?  I think we've all agreed to drop staging, so let's do that.

**Comment by murilovmachado** [comment]
@nickhurlburt Sure, I'll do that right now.
